### ISSUE TITLE
Wishlist for Autotune

### ISSUE THREAD

**Issue by Bender1061** [openaps/oref0#627]
Something that would be real nice to have is the ability of Autotune to use different basal Patterns.

So when autotune is run, it would take in account what Pattern was running during the times it's adjusting.

so then it would use the current Pattern with the current Autotune pattern to make autotune adjustments.
ie you could have a recommendations for Default Pattern and Pattern A and B  

Yes; for some people Autosense should be able to account for it, but there is a big difference fro me when I'm very busy and when I'm being a couch/desk potato. 

I don't like using Patterns as much as I used too now, because Autotune is not really accurate if I change it (at least for the next day).

On top of that, if there was a way to change what Patterns we were using with OpenAPS (I'm not sure if we have that ability or not) then We could make a calendar if I know it's going to be an active day (think of kids with sports) and that could trigger the change of Pattern using IFTTT. 

just a couple things that would be real nice if we could do that.  Anyone Bored?

**Comment by PieterGit** [comment]
I have thought that it would be maybe possible to also register activity during the day and take that into account for autotune and/or autosense and/or (temp) targets. Calculate the  steps made in previous 15 minutes (calculated by the Urchin app on the Pebble). If there has been activity in the past 15 minutes, adjust targets / autotune / autosense temporarily.

There are a number of potential problems with that:
- it will raise it after the activity, whereas it would be better to set it before the activity
- only works for steps count (which is quite inaccurate), and not e.g. for swimming
- I didn't know how to calculate the impact of step count on glucose levels in the future, and so it may be in conflict with the reference design: "OpenAPS does not have any complicated machine learning algorithms, 17-parameter insulin response models, or differential equations. As such, OpenAPS system is open and transparent in how it works, and understandable not just by experts, but also by clinicians and end users (patients)."

**Comment by scottleibrand** [comment]
@PieterGit I think https://github.com/openaps/oref0/pull/684 provides the mechanism we'd want to use for adjusting real-time loop operation for exercise.  If you wanted to integrate something to set temp targets automatically based on observed activity, that might be a worthwhile enhancement, but I think Exercise Mode will be most useful when set proactively before starting activity.

@Bender1061 I'm inclined to work more on making autosens automatically detect and properly respond to the impact of activity levels, rather than trying to autotune the different patterns differently.  In addition to Exercise Mode, there is also a still-open PR to do something similar with autosens' sensitivityRatio.  Maybe as a next action we should look into whether those features, along with appropriate use of temp targets to inform the system of the onset of high-activity periods, can appropriately handle the situation you're describing?

**Comment by danamlewis** [comment]
I've been pleased with the 0.6.0 exercise mode settings to accomplish this goal. @Bender1061, have you tried those? Should we keep this issue open, or close if you feel those are meeting this need?

**Comment by danamlewis** [comment]
Closing due to lack of inactivity for 1.5y; please re-open with any notes about testing the exercise mode settings if this needs additional improvements for other use cases. Thanks!
